### ISSUE TITLE
mushroom boss needs to have several adjustments for balance

### ISSUE THREAD

**Issue by rkettering** [frogatto/frogatto#152]
Right now the whole mechanic behind the homing shots is a bit broken in several ways:
- You're expected to trick them into hitting solid terrain to leave behind a "shot" you can fire back
- but said solid terrain severely impedes your movement
- and very frequently the upward "blast" attacks prevent you from dodging, either

We probably want to ditch the "solid terrain" thing entirely, and move the shots to being "shot out of the air" instead.  We may consider dropping the number of shots to one, and/or making the shots no longer be a source of ammo - making the mushroom boss spawn little mushroom_jellyfish as the source of ammo instead (this would have the bonus of allowing some amount of refueling stuff to drop from them, making the fight less of a brutal endurance match).

**Comment by Phanterm** [comment]
I like the idea of the shots themselves being the ammo. I would prefer this option over the jellyfish spawner, because we could make the shots have a certain amount of 'hitpoints' (say, 3-5 hits from an energy shot) before losing its harmful glow, falling to the ground, and becoming swallow-able as a source of ammo.

As for the terrain, given the aforementioned option, perhaps we may also consider the use of ethereal orbs to shape the battlefield relative to the player's position in the boss room? I know that those elements are meant for the alteborea, but...

**Comment by rkettering** [comment]
We've done a few updates tonight; we've limited the homing shots to only one at a time, and they no longer spawn ammo.  We've added spawning jellyfish mushrooms as a spawned shot.

Finally, we've placed wooden bridges in place of the obstructing overhead rock (which was really there only as a way to kill the homing shots).  This offers a lot more dodging flexibility to the player, which was one of the primary impetuses to this redo.

I'm not against setting up the homing shots to be killable via player attacks, though right now they present a unique challenge because the player can kill most enemy attacks with his own shots (in fact some of his own will 'barrel through' multiple enemy shots to hit a target, like the homing shot, and it's a bit of a nice "turn the tables" trick to reverse that on the player - to have an enemy shot that instead barrels through the player's shots.

We should now playtest this level before doing any further changes.
